core/linux.go

```diff
@@ -755,6 +757,9 @@ func (*linuxGenerator) install(m interface{}, ctx blueprint.ModuleContext) []str
 
 	for _, src := range ins.filesToInstall(ctx) {
 		dest := filepath.Join(installPath, filepath.Base(src))
+		if !strings.HasPrefix(src, g.buildDir()) {
```

Comment:
Is it possible to do this at the module type level?
- `bob_resource` always installs things from the source dir
- Every other module type always installs from the build dir

We should be able to do this reliably that way, rather than guessing based on the existing prefix

Reply:
Prefixing it at higher level is not possible due to the interference with the Android but it's possible to check type, so I've changed it. Also I've introduced the comment for further future, so it will be more obvious what's being done